File: src/main/java/me/mattstudios/triumphpets/pet/PetController.java
package me.mattstudios.triumphpets.pet;

import org.bukkit.Location;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;

public interface PetController {

    /**
     * Spawns a pet in the world.
     *
     * @param location The location the pet will appear.
     * @param player   The player to be the owner of the pet.
     */
    void spawnPet(Location location, Player player);

    /**
     * Checks weather or not the entity that is being tracked is a pet or not.
     *
     * @param entity The entity to check.
     * @return True if it is a pet, false if it is not.
     */
    boolean isPet(Entity entity);

    /**
     * Removes all the teams created in process and all the pet entities spawned.
     */
    void removeAll();

}

File: common/src/main/kotlin/me/mattstudios/triumphpets/pet/PetController.kt
package me.mattstudios.triumphpets.pet

import me.mattstudios.triumphpets.data.PetData
import org.bukkit.Location
import org.bukkit.entity.Entity
import org.bukkit.entity.Player

/**
 * @author Matt
 */
interface PetController {

    /**
     * Spawns the pet
     */
    fun spawnPet(petPlayer: PetPlayer, petData: PetData)

    /**
     * Despawns the pet from player
     */
    fun despawnPet(player: Player)


    /**
     * Checks if the entity is a pet or not
     */
    fun isPet(entity: Entity): Boolean

    /**
     * Checks if the entity is a pet component
     */
    fun isPetComponent(entity: Entity): Boolean

    /**
     * Spawns a new pet on the location
     */
    //fun spawnPet(location: Location, player: Player)

    /**
     * Removes all the pets from the world
     */
    fun removeAll()

    fun removeCrash()
}